sink_routing_test.go

```diff
@@ -338,3 +338,42 @@ match:
 	assert.True(t, config.Match("aa", []string{"ab", "baab"}))
 	assert.False(t, config.Match("bb", []string{"ab", "baab"}))
 }
+
+func TestUnmarshallingAndMarshalling(t *testing.T) {
+	config := veneur.SinkRoutingConfig{}
+	yamlString := []byte(`name: test
+match:
+  - name:
+      kind: exact
+      value: aa
+    tags:
+      - kind: exact
+        value: ab
+  - name:
+      kind: exact
+      value: bb
+    tags:
+      - kind: prefix
+        value: aa
+        unset: true 
+`)
+
+	require.Nil(t, yaml.Unmarshal(yamlString, &config))
+
+	backToYaml, err := yaml.Marshal(config)
+	require.Nil(t, err)
+
+	nConfig := veneur.SinkRoutingConfig{}
```

Comment:
nit: you probably want to assert on the `backToYaml` value, not re-marshal it again.